src/java/main/org/apache/zookeeper/server/auth/ProviderRegistry.java

```diff
@@ -61,17 +64,17 @@ public static void initialize() {
         }
     }
 
-    public static AuthenticationProvider getProvider(String scheme) {
+    public static ServerAuthenticationProvider getProvider(ZooKeeperServer zks, String scheme) {
```

Comment:
We can un-deprecate the existing `AuthenticationProvider` and let people choose which API they want.